### PR TITLE
331 combine extended procedures branches and test before merging to dev

### PR DESCRIPTION
**Pull request by Donnie-Ice**

### CODE REVIEW

**Review by rjbrown2 — APPROVED** [review]

File: .gitignore
Comment (jlucas9) [inline .gitignore:1]: Looks like some extra files may have been ignored here that we may want, like the top level Makefile. Can we more to ignore: bin/* and */CMakeFiles/*?

File: src/core/crypto_key_mgmt.c [hunk 1]
```diff
@@ -44,13 +44,17 @@ int32_t Crypto_Key_OTAR(void)
     int x = 0;
     int y;
     int32_t status = CRYPTO_LIB_SUCCESS;
-    int pdu_keys = (sdls_frame.pdu.pdu_len - 30) / (2 + KEY_SIZE);
+    // uint16_t pdu_len = (uint16_t) sdls_frame.pdu.hdr.pdu_len[1] << 8 | sdls_frame.pdu.hdr.pdu_len[0];
+    int pdu_keys = (sdls_frame.pdu.hdr.pdu_len - 30) / (2 + 32);
```
Comment (jlucas9) [inline src/core/crypto_key_mgmt.c:48]: It is safe to assume KEY_SIZE is static at 32 bytes now?

File: src/core/crypto_key_mgmt.c [hunk 2]
```diff
@@ -65,22 +69,28 @@ int32_t Crypto_Key_OTAR(void)
...
         }
+#ifdef DEBUG
         printf(KRED "Error: MKID is not valid! \n" RESET);
+#endif
         status = CRYPTO_LIB_ERROR;
         return status;
     }
 
-    for (count = 2; count < (2 + IV_SIZE); count++)
+    for (count = 2; count < (2 + 12); count++)
```
Comment (jlucas9) [inline src/core/crypto_key_mgmt.c:79]: Is it safe to assume IV_SIZE is static at 12 now?

File: src/sa/internal/sa_interface_inmemory.template.c [hunk 3]
```diff
@@ -1510,12 +1521,17 @@ static int32_t sa_setARSNW(void)
 
         for (x = 0; x < sa[spi].arsnw_len; x++)
         {
-            sa[spi].arsnw = (((uint8_t)sdls_frame.pdu.data[x + 3]) << (sa[spi].arsnw_len - x));
+            sa[spi].arsnw = (((uint8_t)sdls_frame.pdu.data[x + 2]));
```
Comment (jlucas9) [inline src/sa/internal/sa_interface_inmemory.template.c:1524]: I'm not sure this makes sense as a for loop as it continually will overwrite the arsnw

File: support/scripts/update_env.sh
Comment (jlucas9) [inline support/scripts/update_env.sh:1]: Do we need this script still if we're using a fixed prepared docker container?

File: src/sa/internal/sa_interface_inmemory.template.c (hunk 3 above)
Comment (Donnie-Ice) [reply src/sa/internal/sa_interface_inmemory.template.c:1524]: Correct, I fixed the shift but forgot to take the loop out

File: include/crypto_config.h [hunk 4]
```diff
@@ -205,6 +205,12 @@
 #define AOS_FRAME_DATA_SIZE 1786 /* bytes */
 #define AOS_FILL_SIZE 1145       /* bytes */
 
+// SDLS Behavior Defines
+#define SDLS_KEYV_MAX_KEYS 21  /* keys */
+#define SDLS_IV_LEN 12         /* bytes */
+#define SDLS_KEYV_KEY_ID_LEN 2 /* bytes */
+#define SDLS_KEY_LEN 32        /* bytes */
+
```
[the remark this answers is not in the document]
Comment (Donnie-Ice) [reply include/crypto_config.h:213]: I don't think `IV_LEN 12` will apply to the entirety of CryptoLib. I think that the `KEY_LEN 32` will currently apply to CryptoLib, but may not in the future. For `KEY_ID_LEN 2`, I think it makes sense to specify that it is for SDLS, but may be able to remove the `KEYV` part as it is used in multiple EP functions. Could be wrong on these assumptions.

File: support/scripts/update_env.sh
Comment (Donnie-Ice) [reply support/scripts/update_env.sh:1]: It is being used as part of the CodeQL workflow. I'm not sure that it is absolutely needed, but it does make the workflow file look nicer.

File: include/crypto_config.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (Donnie-Ice) [reply include/crypto_config.h:213]: To be referenced in new issue

File: src/core/crypto_key_mgmt.c (hunk 2 above)
Comment (Donnie-Ice) [reply src/core/crypto_key_mgmt.c:79]: To be referenced in new issue

File: src/core/crypto_key_mgmt.c (hunk 1 above)
Comment (Donnie-Ice) [reply src/core/crypto_key_mgmt.c:48]: To be referenced in new issue

File: src/core/crypto_tc.c
Comment (dccutrig) [inline src/core/crypto_tc.c:1]: Circa 1886 we have a TODO :(

**Review by dccutrig — APPROVED** [review]

**Review by jlucas9 — APPROVED** [review]

**Pull request merged**